impl/src/main/java/org/jboss/weld/bean/ManagedBean.java

```diff
@@ -168,10 +172,13 @@ public T create(CreationalContext<T> creationalContext) {
             RequestContext context = getUnboundRequestContext();
             try {
                 context.activate();
+                FastEvent.of(Object.class, beanManager, Initialized.Literal.REQUEST).fire(this.getId());
```

Comment:
We need to optimize here (cache the event instance or something like this) - this would probably degrade the performance significantly.